pkg/sql/alter_table.go

```diff
@@ -1807,56 +1682,37 @@ func dropColumnImpl(
 		}
 	}
 
-	// Drop unique constraints that reference the column.
-	sliceIdx := 0
-	for i := range tableDesc.UniqueWithoutIndexConstraints {
-		constraint := &tableDesc.UniqueWithoutIndexConstraints[i]
-		tableDesc.UniqueWithoutIndexConstraints[sliceIdx] = *constraint
-		sliceIdx++
-		if descpb.ColumnIDs(constraint.ColumnIDs).Contains(colToDrop.GetID()) {
-			sliceIdx--
-
-			// If this unique constraint is used on the referencing side of any FK
-			// constraints, try to remove the references. Don't bother trying to find
-			// an alternate index or constraint, since all possible matches will
-			// be dropped when the column is dropped.
-			if err := params.p.tryRemoveFKBackReferences(
-				params.ctx, tableDesc, constraint, t.DropBehavior, nil,
-			); err != nil {
-				return nil, err
-			}
+	// Drop non-index-backed unique constraints which reference the column.
+	for _, uwoi := range tableDesc.EnforcedUniqueConstraintsWithoutIndex() {
+		if uwoi.Dropped() || !uwoi.CollectKeyColumnIDs().Contains(colToDrop.GetID()) {
+			continue
+		}
```

Comment:
I think (but I'm not sure) that omitting dropped unique-without-indexes was a bug.